superduperdb/backends/base/query.py

```diff
@@ -1,761 +1,268 @@
-            components.append(str(self.pre_like))
-        if self.query_linker:
-            components.extend(self.query_linker.repr_().split('.')[1:])
-        if self.post_like:
-            components.append(str(self.post_like))
-        return '.'.join(components)
-
-    @classmethod
-    def _query_from_parts(
-        cls,
-        table_or_collection,
-        pre_like=None,
-        post_like=None,
-        query_linker=None,
-    ):
-        return cls(
-            table_or_collection=table_or_collection,
-            pre_like=pre_like,
-            post_like=post_like,
-            query_linker=query_linker,
-        )
-
-    def _get_query_component(
-        self,
-        name: str,
-        type: str,
-        args: t.Optional[t.Sequence] = None,
-        kwargs: t.Optional[t.Dict] = None,
-    ):
-        query_component_cls = self.table_or_collection.query_components.get(
-            name, QueryComponent
-        )
-        return query_component_cls(name, type=type, args=args, kwargs=kwargs)
-
-    @abstractmethod
-    def _get_query_linker(cls, table_or_collection, members) -> 'QueryLinker':
-        pass
-
-    def __getattr__(self, name):
-        _check_illegal_attribute(name)
-        assert self.post_like is None
-        if self.query_linker is not None:
-            query_linker = getattr(self.query_linker, name)
+    def unique_id(self):
+        return self.id
+
+    def _deep_flat_encode(self, cache, blobs, files, leaves_to_keep=()):
+        query, documents = self._dump_query()
+        cache[self.id] = {
+            '_path': 'superduperdb.backends.base.new_query.parse_query',
+            'documents': documents,
+            'query': query,
+        }
+        return self.id
+
+    @staticmethod
+    def _update_item(a, documents, queries):
+        if isinstance(a, Query):
+            a, sub_documents, sub_queries = a._to_str()
+            documents.update(sub_documents)
+            queries.update(sub_queries)
+            id_ = uuid.uuid4().hex[:5].upper()
+            queries[id_] = a
+            arg = f'query[{id_}]'
         else:
-            query_linker = self._get_query_linker(
-                self.table_or_collection,
-                members=[self._get_query_component(name, type=QueryType.ATTR)],
+            id_ = uuid.uuid4().hex[:5].upper()
+            if isinstance(a, dict):
+                documents[id_] = a
+                arg = f'documents[{id_}]'
+            elif isinstance(a, list):
+                documents[id_] = {'_base': a}
+                arg = f'documents[{id_}]'
+            else:
+                try:
+                    arg = json.dumps(a)
+                except Exception:
+                    documents[id_] = {'_base': a}
+                    arg = id_
+        return arg
+
+    def _to_str(self):
+        documents = {}
+        queries = {}
+        out = self.identifier[:]
+        for part in self.parts:
+            if isinstance(part, str):
```

Comment:
This will be true only for `table.select().filter(table.id == 123)` `table` part right?